src/summary_result.rs

```diff
@@ -0,0 +1,224 @@
+use std::cmp::Ordering;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::iter::FromIterator;
+use std::string::String;
+
+use crate::protobuf::summary_result::Context;
+use crate::*;
+
+/// Result from calling [summary].
+/// Where possible, this is API-compatible with [ParseResult].
+#[derive(Debug, PartialEq)]
+pub struct SummaryResult {
+    pub protobuf: protobuf::SummaryResult,
+    pub warnings: Vec<String>,
+    pub tables: Vec<Table>,
+    pub aliases: HashMap<String, String>,
+    pub cte_names: Vec<String>,
+    pub functions: Vec<Function>,
+    pub filter_columns: Vec<FilterColumn>,
+    pub truncated_query: Option<String>,
+    pub statement_types: Vec<String>,
+}
+
+impl PartialOrd for FilterColumn {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for FilterColumn {
+    fn cmp(&self, other: &Self) -> Ordering {
+        let schema_cmp = self.schema_name.cmp(&other.schema_name);
+        let table_cmp = self.table_name.cmp(&other.table_name);
+        let column_cmp = self.column.cmp(&other.column);
+
+        if schema_cmp != Ordering::Equal {
+            schema_cmp
+        } else if table_cmp != Ordering::Equal {
+            table_cmp
+        } else {
+            column_cmp
+        }
```

Comment:
This code looks to be identical to `#[derive(PartialOrd, Ord)]`, which sorts a struct by the order that its fields are defined.